pkg/lib/journal.js

```diff
@@ -211,7 +211,7 @@ journal.printable = function printable(value) {
     else if (typeof (value) == "string")
         return value;
     else if (value.length !== undefined)
-        return cockpit.format(_("[$0 bytes of binary data]"), value.length);
+        return new TextDecoder().decode(new Uint8Array(value));
```

Comment:
Sorry, I just thought about this again, and this is too simplistic. E.g. `trec` (cockpit-session-recording) also puts its data into the journal. It could be the case that there is large binary data that would overwhelm the page. Also, the journal explicitly supports `MESSAGE=` vs. `BIN=` fields, you can see this with `journalctl -o verbose`:
```
printf 'MESSAGE=Hello \01 World\nPRIORITY=3\nFOO=bar\nBIN=a\01b\02c\03\n' | logger --journald

journalctl -o verbose --since '5 minutes ago'
    MESSAGE=[13B blob data]
    FOO=bar
    BIN=[6B blob data]
```

So `journalctl` without verbose only shows "[13B blob data]", and cockpit should also do that. It's *only* legit to decode that if the binary data is in the `MESSAGE=` field.

Reply:
so all we have to do is decode the binary data in the MESSAGE field.
Hints would be nice ツ

Reply:
Ah, it's actually fine:
```js
message: journal.printable(journal_entry.MESSAGE)
```
So we already don't show `BIN` in the "log overview" page. I suppose what confused me was your test case change for BIN=, see below.